src/logic/treeBuilder/treeBuilder.ts

```diff
@@ -0,0 +1,24 @@
+import * as _ from "lodash";
+import { TreeNode } from "./treeNode";
+
+export interface TreeBuildParams {
+  branchFactor: number; // number of childrens for each node
+  treeDepth: number; // number of levels except root
+}
+
+export class TreeBuilder {
+  static build({ branchFactor, treeDepth }: TreeBuildParams): TreeNode {
+    if (treeDepth === 1) {
+      const subtree = _.range(0, branchFactor).map(() => {
+        const nodeReward = Math.random() - 0.5; // js random return values from 0 to 1, so here I'm mapping it to appropriate range
```

Comment:
Chyba nie rozumiem po co to mapowanie.

Reply:
Bo może być ujemny `reward` kiedy przegramy partię.
A random w js zwraca wartości od 0 do 1, więc nie ma możliwości uzyskać ujemną wartość